ste/downloader-azureFiles_windows.go

```diff
@@ -78,6 +81,33 @@ func (*azureFilesDownloader) PutSDDL(sip ISMBPropertyBearingSourceInfoProvider,
 		return fmt.Errorf("parsing SDDL: %s", err)
 	}
 
+	ctl, _, err := sd.Control()
+	if err != nil {
+		return fmt.Errorf("getting control bits: %w", err)
+	}
+
+	var securityInfoFlags windows.SECURITY_INFORMATION = windows.OWNER_SECURITY_INFORMATION | windows.GROUP_SECURITY_INFORMATION | windows.DACL_SECURITY_INFORMATION
+
+	// remove everything down to the if statement to return to xcopy functionality
+	// Obtain the destination root and figure out if we're at the top level of the transfer.
+	plan := a.jptm.Plan()
+	destRoot := string(plan.DestinationRoot[:plan.DestinationRootLength])
+	relPath, err := filepath.Rel(destRoot, txInfo.Destination)
+
+	if err != nil {
+		// This should never ever happen.
+		panic("couldn't find relative path from root")
+	}
+
+	// Golang did not cooperate with backslashes with filepath.SplitList.
+	splitPath := strings.Split(relPath, common.DeterminePathSeparator(relPath))
+
+	// Protected ACLs see no inheritance whatsoever.
+	// remove the second half of the if statement to return to xcopy functionality
```

Comment:
Needs a comment here to explain _why_ we are doing the len(splitPath) == 1.  It took us hours, and plenty of discussion to figure out that we should do that, and why. So we need to document that fact for future readers.

Ideally I'd suggest, make a variable, called something like isAtDestinationRoot, and assign len(splitPath) to that variable, then use the varaible in the IF. And even another var called, `isProtectedAtSource :=  (ctl&windows.SE_DACL_PROTECTED) != 0`, and use that in the variable too. Then there won't be much left that you need to explain in the comment. It will only need to say that we protect everything that's at the root level, to get "RoboCopy-like" behavior. I.e. preserve all permissions from source, and inherit no new ones from destination.